wagtail/admin/views/chooser.py

```diff
@@ -298,22 +298,23 @@ def get(self, request, parent_page_id=None):
 
                 # we are at the Root level, so get the locales from the current pages
                 choose_url = reverse("wagtailadmin_choose_page")
-                locale_options = [
-                    {
-                        "locale": locale,
-                        "url": choose_url
-                        + "?"
-                        + urlencode(
-                            {
-                                "page_type": page_type_string,
-                                "locale": locale.language_code,
-                            }
-                        ),
+                for locale in Locale.objects.filter(
+                    pk__in=pages.values_list("locale_id")
+                ).exclude(pk=active_locale_id):
+                    params_dict = {
+                        "page_type": page_type_string,
+                        "locale": locale.language_code,
+                        "user_perms": user_perm,
                     }
-                    for locale in Locale.objects.filter(
-                        pk__in=pages.values_list("locale_id")
-                    ).exclude(pk=active_locale_id)
-                ]
+                    if not user_perm:
+                        del params_dict["user_perms"]
```

Comment:
Thanks for highlighting the issue! It seems this is not the only query parameter that gets lost during the locale switching. Instead of specifying each query param (and potentially forgetting to do this for future query params), we could copy the current request's query params and remove ones that aren't necessary. I noticed that in other places we tend to do this, e.g.:

https://github.com/wagtail/wagtail/blob/dd5e16b2e19c63c0764da8d5f2a16661386bf73f/wagtail/admin/templates/wagtailadmin/chooser/_browse_results.html#L18

https://github.com/wagtail/wagtail/blob/dd5e16b2e19c63c0764da8d5f2a16661386bf73f/wagtail/admin/templates/wagtailadmin/chooser/tables/page_navigate_to_children_cell.html#L4